CheckoutKata/Checkout.cs

```diff
@@ -3,18 +3,41 @@ namespace CheckoutKata;
 public class Checkout
 {
     private readonly ISkuRetrievalService _skuRetrievalService;
+    private readonly ISkuSpecialRetrievalService _skuSpecialRetrievalService;
     
-    public Checkout(ISkuRetrievalService skuRetrievalService)
+    public Checkout(ISkuRetrievalService skuRetrievalService, ISkuSpecialRetrievalService skuSpecialRetrievalService)
     {
         _skuRetrievalService = skuRetrievalService;
+        _skuSpecialRetrievalService = skuSpecialRetrievalService;
     }
     
     public int Total { get; private set; }
+    
+    private readonly List<string> _skuCodes = new();
 
     public void Scan(string code)
     {
         var validSku = _skuRetrievalService.GetSkuByCode(code);
+        
+        _skuCodes.Add(validSku.Code);
+        
+        Console.WriteLine(_skuCodes);
+
+        
+        var skuSpecial = _skuSpecialRetrievalService.GetSkuSpecialByCode(code);
```

Comment:
You've got 2 concepts that are kind of bound here now.  Sku and Sku special.
Personally would likely have Specials or Discounts as a component of Sku we can prolly look at something for that because you mentioned something about scaling additional types and I think it's something worth exploring